packages/server/lib/plugins/child/ts_node.js

```diff
@@ -3,18 +3,39 @@ const debugLib = require('debug')
 const path = require('path')
 const tsnode = require('ts-node')
 const resolve = require('../../util/resolve')
+const semver = require('semver')
 
 const debug = debugLib('cypress:server:ts-node')
 
 const getTsNodeOptions = (tsPath, registeredFile) => {
+  const version = require(tsPath).version || '0.0.0'
+
+  /**
+   * NOTE: This circumvents a limitation of ts-node
+   *
+   * The "preserveValueImports" option was introduced in TypeScript 4.5.0
+   * https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-5.html#disabling-import-elision
+   *
+   * If we pass an unknown compiler option to ts-node,
+   * it ignores all options passed.
+   * If we want `module: "commonjs"` to always be used,
+   * we need to only set options that are supported in this version of TypeScript.
+   */
+  const compilerOptions = {
+    module: 'commonjs',
+    ...(semver.satisfies(version, '>=4.5.0')
+      // Only adding this option for TS >= 4.5.0
+      ? { preserveValueImports: false }
```

Comment:
@elevatebart why do we need to pass this option in child processes?

Reply:
this needs to be passed every where we compile ts-code from a user.
It can have a different version of TypeScript and a different `tsconfig.json` than what we have internally.

Reply:
i'm still not clear why this is needed

Reply:
If the user has TypeScript beyond 4.5 and has this tsconfig

```json
{
  "compilerOptions": {
    "preserveValueImports": true,
    "module": "es2015"
  }
}
```

To have it work as close as possible of their config we will use their version of TypeScript and their tsconfig.json
We need to make it work with ts-node though, so we will change the "module" to be `commonjs`. 
Unfortunately this configuration makes TypeScript throw an error:

`preserveValueImports` can only be set with `module` at least `es2015`.

To circumvent this error (described in the ticket) we force the value to false.

BUT... ts-node does not accept any unknown values in the config so if we pass it all the time, it will fail in older versions of typescript that do not know this option. 

We protect it by only adding the option when we know it exists.